hugr-core/src/ops/constant.rs

```diff
@@ -326,14 +326,15 @@ pub enum ConstTypeError {
 
 /// Hugrs (even functions) inside Consts must be monomorphic
 fn mono_fn_type(h: &Hugr) -> Result<FunctionType, ConstTypeError> {
```

Comment:
this function just matches existing behaviour - this PR does not address the question of what a valid monomorphic function value hugr root type is